src/main/java/net/fabricmc/loom/configuration/providers/mappings/IntermediaryMappingsProvider.java

```diff
@@ -45,24 +55,59 @@ public abstract class IntermediaryMappingsProvider extends IntermediateMappingsP
 	public abstract Property<Boolean> getRefreshDeps();
 
 	@Override
-	public void provide(Path tinyMappings) throws IOException {
+	public void provide(Path tinyMappings, Project project) throws IOException {
 		if (Files.exists(tinyMappings) && !getRefreshDeps().get()) {
 			return;
 		}
 
 		// Download and extract intermediary
 		final Path intermediaryJarPath = Files.createTempFile(getName(), ".jar");
 		final String encodedMcVersion = UrlEscapers.urlFragmentEscaper().escape(getMinecraftVersion().get());
-		final String url = getIntermediaryUrl().get().formatted(encodedMcVersion);
+		final String urlRaw = getIntermediaryUrl().get();
 
-		LOGGER.info("Downloading intermediary from {}", url);
+		if (project != null && urlRaw.equals(LoomGradleExtensionApiImpl.DEFAULT_INTERMEDIARY_URL)) {
+			final Configuration config = project.getConfigurations()
```

Comment:
A deteached configuration should work here

Reply:
The idea behind using `defaultDependencies` and a named configuration was to allow specifying a custom artifact in the build script.

Reply:
Humm, I see. Im not sure we need a 3rd way to change where intermediary comes from, its not something people do often.